File: Cpp/033.cpp
#include<iostream>
#include<cmath>
using namespace std;

double cancel(int a,int b)
{
    int p=a/10,q=a%10,r=b/10,s=b%10;
    if(p==r) return (double)(q)/s;
    else if(q==r) return (double)(p)/s;
    else if(p==s) return (double)(q)/r;
    else if(q==r) return (double)(p)/s;
    else return 0.0;
}

int main()
{
    int num,den;
    double prod=1;
    for(int a=1;a<99;a++)
    {
        for(int b=a+1;b<=99;b++)
        {
            double c=(double)(a)/b;
            double d=cancel(a,b);
            if(c==d) prod*=c;
        }
    }
    cout<<prod;
}

File: Ruby/033.rb
MAX = 99

def cancel(a, b)
  common = (a.to_s.chars & b.to_s.chars)
  return 0 if common.empty? || common.include?('0')
  # puts a.to_s.delete(common)
  # puts b.to_s.delete(common)
  common = common.to_s
  a.to_s.delete(common).to_f / b.to_s.delete(common).to_f
end

ans = []
1.upto(MAX - 1).each do |num|
  (num + 1).upto(MAX).each do |den|
    val = num.to_f / den
    if val == cancel(num, den)
      ans << "#{num}/#{den}"
    end
  end
end

puts ans